src/main/webapp/go.js

```diff
@@ -299,19 +340,39 @@ function sendIndexToServlet(index) {
  * Disable lint check because proceed() is called from go.html.
  */
 function proceed() { //eslint-disable-line
-  destIndex++;
-  sendIndexToServlet(destIndex);
-  if (destIndex < huntArr.length) {
-    updateMessage(RIDDLE_DISPLAY, 'Riddle: ' + huntArr[destIndex].puzzle);
-    toggleHintButton(/* hide = */ false);
+  hunt.destIndex = hunt.destIndex + 1;
```

Comment:
Optional: I think it would be nice to bundle this up into a method on the manager class, e.g. `incrementDestIndex()`, or you could even bundle in the accessor part that happens on the next line with `incrementAndGetDestIndex()`.